### DISCUSSION TITLE
UI Components testing using Lighthouse or similar tools

### DISCUSSION THREAD

**Discussion by shelgeram07** [SigNoz/signoz#208] [Ideas]
Along with all the functional testing of application, its also equally important to have a check on frontend code performance and optimization areas on a regular basis. Ideally, there should be an automated job to run performance audits for any new UI component before it gets merged into master.  Google lighthouse can be a good candidate to incorporate in overall release/ CI/CD process to ensure, we are not introducing any performance degradation. We can leverage lighthouse cli to automate this process. 
Please let me know if you want to have more detailed discussion on this to discuss on approach, implementation and metrics to measure etc.

**Comment by pranay01** [comment]
Hey @shelgeram07 thanks for the note. Can you share some more details on how to integrate lighthouse CLI tools in the release/PR review process? If you have any blogs/write ups you can share - that would be great. If you know any other open source projects which do this, would be good to learn from them also

I have used lighthouse web tool earlier but don't have experience incorporating it into build/release pipelines.

**Reply by shelgeram07** [reply]
@pranay01  I had implemented lighthouse auditing as part of release process earlier and would be able to help on setting this up. Actually, let me come up with a blog / write up for same which we can together evolve as per our needs. Thoughts?
